src/native/corehost/setup.cmake

```diff
@@ -1,44 +1,52 @@
 # Licensed to the .NET Foundation under one or more agreements.
 # The .NET Foundation licenses this file to you under the MIT license.
 
-if (CLI_CMAKE_PORTABLE_BUILD)
-    add_definitions(-DFEATURE_PORTABLE_BUILD=1)
-endif ()
-
-if("${CLI_CMAKE_HOST_POLICY_VER}" STREQUAL "")
-    message(FATAL_ERROR "Host policy version is not specified")
+if(${CLR_CMAKE_BUILD_SUBSET_RUNTIME})
+    # CLR partition builds only the single file host where hosting components are all statically linked.
+    # the versioning information is irrelevant and may only come up in tracing.
+    # so we will use "static"
+    add_definitions(-DHOST_POLICY_PKG_VER="static")
+    add_definitions(-DHOST_FXR_PKG_VER="static")
+    add_definitions(-DHOST_PKG_VER="static")
+    add_definitions(-DCOMMON_HOST_PKG_VER="static")
+    add_definitions(-DHOST_POLICY_PKG_NAME="static")
+    add_definitions(-DHOST_POLICY_PKG_REL_DIR="static")
+    add_definitions(-DREPO_COMMIT_HASH="static")
```

Comment:
I wonder if this change won't add "static" as another commit hash to the host (besides the one coming from coreclr). What does `strings singlefilehost | grep "@(#)"` show? It should print just one string with the real commit hash.

Reply:
This is just a string that could be seen in tracing. 
Standalone host also has `hostfxr_get_dotnet_environment_info` API, but singlefile host does not expose hosting APIs. The final stitched together executable is not supposed to act as a host - it is just a user program. 

---
In reply to: [578796864](https://github.com/dotnet/runtime/pull/48254#discussion_r578796864) [](ancestors = 578796864)

Reply:
The format of this string literal is like "a874f3adcb3f159edb077ea9bb7d841391ed7465"  - no `#`

---
In reply to: [578823880](https://github.com/dotnet/runtime/pull/48254#discussion_r578823880) [](ancestors = 578823880,578796864)

Reply:
Got it, I've thought it is used by the version string stored in our executables.